svc-lighthouse-api/src/test/java/gov/va/vro/abddataaccess/service/FhirClientTest.java

```diff
@@ -61,55 +84,43 @@ class FhirClientTest {
   private static final String CONDITION_RESPONSE = "condition-response-bundle.json";
   private static final String PROCEDURE_RESPONSE = "procedure-response-bundle.json";
+  private static final String EMPTY_RESPONSE = "empty-bundle.json";
 
   private static final int DEFAULT_PAGE = 1;
   private static final int DEFAULT_SIZE = 30;
-  private static final String FHIRURL = "https://sandbox-api.va.gov/";
+  private static final String FHIR_URL = "https://sandbox-api.va.gov/";
   private static final String TEST_KEY_FILE = "testkey.pem";
   private static final String TEST_TOKEN = "lighthouseToken.json";
 
   @InjectMocks private FhirClient client = Mockito.spy(new FhirClient());
 
-  @Mock private RestTemplate restTemplate;
-
   @Mock private LighthouseApiService lighthouseApiService;
 
+  @Mock private RestTemplate restTemplate;
+
   @Mock private LighthouseProperties properties;
 
   @Mock private IParser jsonParser;
 
-  private IParser parser;
-  private Bundle medicationBundle;
-  private Bundle bpBundle;
-  private String medicationResponseBody;
-  private String bpResponseBody;
+  private static IParser parser;
 
-  @BeforeEach
-  void setUp() throws AbdException, IOException {
+  private static BundleInfo medBundleInfo;
+  private static BundleInfo bpBundleInfo;
+  private static BundleInfo conditionBundleInfo;
+  private static BundleInfo procedureBundleInfo;
+  private static BundleInfo emptyBundleInfo;
+
+  @BeforeAll
+  private static void initVariables() throws IOException {
     FhirContext fhirContext = FhirContext.forR4();
     EncodingEnum respType = EncodingEnum.forContentType(CONTENT_TYPE);
     parser = respType.newParser(fhirContext);
 
-    String testfile =
-        Objects.requireNonNull(getClass().getClassLoader().getResource(MEDICATION_REQUEST_RESPONSE))
-            .getPath();
-    medicationBundle = getMedicalInfoBundle(testfile);
-    medicationResponseBody = new String(Files.readAllBytes(Paths.get(testfile)));
-
-    testfile =
-        Objects.requireNonNull(getClass().getClassLoader().getResource(OBSERVATION_RESPONSE))
-            .getPath();
-    bpBundle = getMedicalInfoBundle(testfile);
-    bpResponseBody = new String(Files.readAllBytes(Paths.get(testfile)));
-  }
-
-  private void mockPropertyNParser() {
-    Mockito.doReturn(FHIRURL).when(properties).getFhirurl();
-
-    Mockito.doReturn(medicationBundle)
-        .when(jsonParser)
-        .parseResource(Bundle.class, medicationResponseBody);
-    Mockito.doReturn(bpBundle).when(jsonParser).parseResource(Bundle.class, bpResponseBody);
+    bpBundleInfo = BundleInfo.getInstance(OBSERVATION_RESPONSE);
+    medBundleInfo = BundleInfo.getInstance(MEDICATION_REQUEST_RESPONSE);
+    conditionBundleInfo = BundleInfo.getInstance(CONDITION_RESPONSE);
+    procedureBundleInfo = BundleInfo.getInstance(PROCEDURE_RESPONSE);
+    emptyBundleInfo = BundleInfo.getInstance(EMPTY_RESPONSE);
   }
```

Comment:
It might be a good choice to use @BeforeAll. Is that caused the private static method to be created? Are all the BundleInfos needed for each test? But, anyway, if they work, they are okay.

Reply:
No, not each  bundle  needed for each test but with BeforeAll we were recreating medication and blood pressure bundles again and again for each test. I thought this was better choice to create once and use when needed.